test/e2e/elasticache/util.py

```diff
@@ -0,0 +1,135 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+"""ElastiCache-specific test utility functions
+"""
+
+import logging
+import boto3
+
+from time import sleep
+from typing import Tuple
+from common import k8s
+
+"""
+Waits for the specified condition in .status.conditions to reach the desired value.
+
+Precondition: resource must be consumed by the controller (i.e. have a .status field)
+
+This differs from some methods in the k8s.py module (e.g. wait_resource_synced) due to
+different representations of conditions across services. For example, that 
+method expects .status.conditions to be a dictionary; for ElastiCache it is
+a list.
```

Comment:
Is this an error in my interpretation of `.status.conditions`? Or is this just the case for ElastiCache? Looking at the generator, it seems that the default is a list?

Reply:
For ElastiCache it's for sure a list of dicts, where each dict has keys `type` (e.g. "ACK.ResourceSynced") and `status` (True/False) among other keys. 

I thought it would be the same across services:
https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/apis/v1alpha1/model.go#L46
https://github.com/aws-controllers-k8s/elasticache-controller/blob/main/apis/v1alpha1/replication_group.go#L71

But I was looking at [this chunk](https://github.com/aws-controllers-k8s/community/blob/main/test/e2e/common/k8s.py#L169-L172), and assumed that Sagemaker had a different structure. Here this method seems to expect `.status.conditions` is a dict, in which `ACK` is a key, whose value is another dict in which `ResourceSynced` is  a key. Let me know what you think

Reply:
@RedbackThomson remember we had this conversation about dict versus list for Conditions and how https://github.com/aws-controllers-k8s/community/blob/7feb390e06c8b862608cc10e561620bbf6e31968/test/e2e/common/k8s.py#L169 was confusing? :)

Reply:
@echen-98 yes, Conditions is a list of dicts. :)

Reply:
Weird. Must have been something I overlooked as part of the initial implementation, or something has since changed. @jaypipes Has it ever been a dictionary?

If `conditions` is always a list, then please override my implementation with your new (correct) one.